interpreter/terminal_interface/terminal_interface.py

```diff
@@ -42,6 +43,22 @@
     # If they don't have readline, that's fine
     pass
 
+def edit_code(code):
+    with tempfile.NamedTemporaryFile(mode='w', delete=False, suffix=".py") as temp_file:
+        temp_file.write(code)
+        temp_filename = temp_file.name
+
+    subprocess.run(['code', temp_filename], check=True)
```

Comment:
Use something like this to open it in default editor. Check language and add extention to the tmp file. python = .py etc..
```
import os
import subprocess
import platform

def open_file_default_editor(file_path):
    try:
        if platform.system() == 'Windows':
            os.startfile(file_path)
        elif platform.system() == 'Darwin':  # macOS
            subprocess.call(('open', file_path))
        elif platform.system() == 'Linux':
            subprocess.call(('xdg-open', file_path))
        else:
            print(f"Unsupported OS: {platform.system()}")
    except Exception as e:
        print(f"Error opening file: {e}")
```

Reply:
Thank you for your feedback!
For checking the language, does OI already know what language the code he wrote is in?

Reply:
Yes, but if language is None it defaults to python

Reply:
Can you please give me the name of the variable that holds that value?

One more thing, I noticed that  whatever changes I made to the code, when I run interpreter in the terminal, it just runs the first version. Why is that?

Reply:
Have a look at contributors.md